internal_controlnet/external_code.py

```diff
@@ -363,17 +287,16 @@ def update_cn_script_in_place(
 
     # fill in remaining parameters to satisfy max models, just in case script needs it.
     max_models = shared.opts.data.get("control_net_unit_count", 3)
-    cn_units = cn_units + [ControlNetUnit(enabled=False)] * max(max_models - len(cn_units), 0)
+    cn_units = cn_units + [UiControlNetUnit(enabled=False)] * max(max_models - len(cn_units), 0)
 
-    cn_script_args_diff = 0
     for script in script_runner.alwayson_scripts:
         if script is cn_script:
-            cn_script_args_diff = len(cn_units) - (cn_script.args_to - cn_script.args_from)
-            script_args[script.args_from:script.args_to] = cn_units
-            script.args_to = script.args_from + len(cn_units)
-        else:
-            script.args_from += cn_script_args_diff
-            script.args_to += cn_script_args_diff
+            for i, unit in enumerate(cn_units):
+                args = unit.construct_args()
+                unit_args_len = min(len(args), script.unit_args_len[i])
+                script_args[script.args_from + script.unit_args_len[i]:unit_args_len] = args[0:unit_args_len]
+                script_args[script.args_from:script.args_to] = cn_units
+                script.args_to = script.args_from + len(cn_units)
```

Comment:
You have already assigned the arguments to the `script_args`, why did you assign cn_units again?